src/ufuncs.c

```diff
@@ -225,8 +225,26 @@ static void Y_Y_func(char **args, npy_intp *dimensions,
 static PyUFuncGenericFunction Y_Y_funcs[1] = {&Y_Y_func};
 
 /* Define the geom, double -> geom functions (Yd_Y) */
-static void *interpolate_data[1] = {GEOSInterpolate_r};
-static void *interpolate_normalized_data[1] = {GEOSInterpolateNormalized_r};
+static void *GEOSInterpolateProtectEmpty_r(void *context, void *geom, double d) {
+    int n = GEOSGeomGetNumPoints_r(context, geom);
+    if (n < 2) {
+        /* return NULL for empty linestrings to protect against a segfault */
```

Comment:
Should this be reported to GEOS, or is it documented that the user needs to check this guarantee?